src/dealer/DealerWallet.ts

```diff
@@ -0,0 +1,515 @@
+import _ from "lodash"
+import { yamlConfig } from "../config"
+import { accountDealerFtxPath, liabilitiesDealerFtxPath } from "../ledger/ledger"
+import { MainBook } from "../mongodb"
+import { OnChainMixin } from "../OnChain"
+import { ILightningWalletUser } from "../types"
+import { btc2sat, sleep } from "../utils"
+import { baseLogger } from "../logger"
+import { UserWallet } from "../userWallet"
+import assert from "assert"
+
+import { GenericExchange, SupportedExchanges, ApiConfig } from "./GenericExchange"
+
+const activeExchangeId: SupportedExchanges = "ftx"
+
+const key = process.env[`${activeExchangeId.toUpperCase()}_KEY`]
+const secret = process.env[`${activeExchangeId.toUpperCase()}_SECRET`]
+const password = process.env[`${activeExchangeId.toUpperCase()}_PASSWORD`]
+
+if (!key || !secret || !password) {
+  throw new Error(`Missing ${activeExchangeId} exchange environment variables`)
+}
+
+const activeApiConfig = new ApiConfig(activeExchangeId, key, secret, password)
+
+// TODO: move to the yaml config
+const simulateOnly = true
+
+export type IBuyOrSell = "sell" | "buy" | null
+
+// FtxDealer is a user of the wallet, therefore we are inheriting from UserWallet
+// FtxDealer is interacting with Ftx through bitcoin layer 1,
+// so it is using the OnChain Mixin.
+export class DealerWallet extends OnChainMixin(UserWallet) {
+  exchange
+
+  constructor({ user, logger }: ILightningWalletUser) {
+    super({ user, logger })
+    this.exchange = new GenericExchange(activeApiConfig)
+    this.logger = logger.child({ topic: "dealer" })
+  }
+
+  async getLocalLiabilities() {
+    // FIXME harmonize the capitalzation for USD/usd
+    const { USD, BTC: satsLnd } = await this.getBalances()
+
+    // TODO: calculate PnL for the dealer
+    // this will influence this account.
+    const { balance: satsFtx } = await MainBook.balance({
+      account: liabilitiesDealerFtxPath,
+      currency: "BTC",
+    })
+
+    return {
+      // dealer is the only one account with the a negative balance for USD
+      // FIXME: look if there is a cleaner design than just have a - sign here
+      usd: -USD,
+      satsLnd,
+      satsFtx,
+    }
+  }
+
+  async has() {
```

Comment:
Can we have a more descriptive name here? Has what exactly?

Reply:
this is a wrapper around a ccxt function. `has()` give the list of function/property the exchange has enabled.

Reply:
Well since we're wrapping the library its a great opportunity to add a more descriptive name so its clearer at the point of usage in our code. Perhaps `exchangeHasProperty` or something like that?